File: Acidaes.MetaData.Rag/DAL/Helper/QueryHelper.cs
namespace Acidaes.MetaData.Rag.DAL.Helper
{
    public static class QueryHelper
    {
        public const string RoleRagDocumentQuery = "select * from RoleView where OwnerId = 101";
        public const string GetObectMastersQuery = "select * from ObjectMasterTableList where ItemType in(1,6,7,8,9)";
        public const string GetObjectListForRagQuery = "select TableName, ItemType, PrimaryField, SubjectField from ObjectMasterTableList where ItemType in(1,6,7,8,9)";
        public const string GetCustomFields = "select Label, FieldId, TableName, KeyId, Type, Mandatory, Description,InternalLabel, FieldName, DisplayType from CustomFieldSchema where OwnerId = 101 and KeyId in(1,6,7,8,9)";
        public const string GetSystemFields = "select Label, FieldId, TableName, KeyId, Type, Mandatory, Description,InternalLabel, FieldName, DisplayType from ObjectSchema where OwnerId = 101 and KeyId in(1,6,7,8,9)";
        public const string GetLayoutWithRolesAndObject = "select distinct lg.ItemTypeID,  lg.LayoutID,ul.Name as LayoutName,lgr.RoleId,Az_Role.Name,lg.LayoutXML from LayoutGroupView lg \r\njoin LayoutGroupRoleMapping lgr on lgr.OwnerId = lg.OwnerID and lgr.LayoutId = lg.LayoutID\r\nand lg.OwnerId = 101 and LayoutType in (2,1) and ItemTypeID in (1,6,7,8,9)\r\njoin UILayoutMaster ul on (ul.OwnerID = lg.OwnerID or ul.OwnerID = 0) and ul.ItemTypeID = lg.ItemTypeID and ul.LayoutID = lg.LayoutID\r\nleft join Az_Role on (Az_Role.AppOwnerID = lgr.OwnerID or Az_Role.AppOwnerID = 0) and lgr.RoleId = Az_Role.RoleID";
    }
}

File: agentic_app/app/database/query_helper.py
class QueryHelper:
    GetRoles = "select * from RoleView"

    GetObjects = "select * from ObjectMasterTableList where ItemType in(1,2,4,5,6,7,8,9,19,34,43,56)"

    GetObjectList = "select TableName, ItemType, PrimaryField, SubjectField from ObjectMasterTableList where ItemType in(1,2,4,5,6,7,8,9,19,34,43,56)"

    GetCustomFields = "select Label, FieldId, TableName, KeyId, Type, Description, FieldName, DisplayType from CustomFieldSchema where OwnerId = 101 and KeyId in(1,2,4,5,6,7,8,9,19,34,43,56)"

    GetSystemFields = "select * from ObjectSchema where OwnerId in (0, 101) and KeyId in(1,2,4,5,6,7,8,9,19,34,43,56)"

    GetLayoutWithRolesAndObject = "select distinct lg.ItemTypeID,  lg.LayoutID,ul.Name as LayoutName,lgr.RoleId, Az_Role.Name,lg.LayoutXML from LayoutGroupView lg join LayoutGroupRoleMapping lgr on lgr.OwnerId = lg.OwnerID and lgr.LayoutId = lg.LayoutID and lg.OwnerId = 101 and LayoutType in (2,1) and ItemTypeID in (1,2,4,5,6,7,8,9,19,34,43,56) join UILayoutMaster ul on (ul.OwnerID = lg.OwnerID or ul.OwnerID = 0) and ul.ItemTypeID = lg.ItemTypeID and ul.LayoutID = lg.LayoutID left join Az_Role on (Az_Role.AppOwnerID = lgr.OwnerID or Az_Role.AppOwnerID = 0) and lgr.RoleId = Az_Role.RoleID"
    
    
    GetLayoutGroupView = "select ItemTypeID, LayoutID, LayoutType, LayoutXML from LayoutGroupView where OwnerId in (0, 101) and LayoutType = 2 and ItemTypeID in(1,2,4,5,6,7,8,9,19,34,43,56)"
    
    GetLayoutUIMaster = "select  LayoutID, ItemTypeID, Name from UILayoutMaster where OwnerId in (0, 101) and ItemTypeID in(1,2,4,5,6,7,8,9,19,34,43,56)"
    
    GetLayoutGroupRoleMapping = "select * from LayoutGroupRoleMapping where OwnerId in (0, 101)"
    
    GetListingFromRptQuery = "select * from RPT_Query where OwnerID in (0,101) and KeyId in (1,2,4,5,6,7,8,9,19,34,43,56)"
    
    GetListingFromGold5Listing = "select * from crmnextlisting_gold5 where OwnerID in (0,101) and RelatedToTypeID in (1,2,4,5,6,7,8,9,19,34,43,56)"
    
    GetListingFromObjectRelationship = "select distinct keyid,listingtype,RelatedKeyID, RelationShipID, RelationShipName from objectrelationship where KeyID in (1,2,4,5,6,7,8,9,19,34,43,56) and RelatedKeyID in (1,2,4,5,6,7,8,9,19,34,43,56)"
